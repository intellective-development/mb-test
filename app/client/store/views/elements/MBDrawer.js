// @flow

import * as React from 'react';
import bindClassNames from '../../../shared/utils/bind_classnames';

import styles from './MBDrawer.scss';

/*
  This component defines the styles for a simple side drawer.
  Its state is intended to be controlled by some parent component, avoiding the need for something like a ref to control it.
*/

const cn = bindClassNames(styles);

type MBDrawerProps = { open: boolean, closeDrawer: () => void, children: React.Node };
const MBDrawer = ({open, closeDrawer, children}: MBDrawerProps) => {
  const drawer_classes = cn('elMBDrawer', {elMBDrawer__Hidden: !open});
  const overlay_classes = cn('elMBDrawer_Overlay', {elMBDrawer_Overlay__Hidden: !open});

  return (
    <div>
      <div className={overlay_classes} onClick={closeDrawer} />
      <div className={drawer_classes}>
        {children}
      </div>
    </div>
  );
};

export default MBDrawer;
