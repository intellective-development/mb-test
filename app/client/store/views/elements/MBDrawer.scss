@import "../style/variables";

$drawer-width: 300px;
$drawer-animate-in-speed: 0.35s;
$drawer-animate-out-speed: 0.2s;

.elMBDrawer {
  position: fixed;
  background-color: $brand-white;
  box-shadow: -5px 0px 5px -5px rgba(0,0,0,0.55);
  overflow-y: scroll;

  top: 0;
  left: 0;
  width: $drawer-width;
  height: 100%;
  z-index: 10;

  transition: all $drawer-animate-in-speed ease-out; // animate in behavior
}
.elMBDrawer__Hidden {
  transition: all $drawer-animate-out-speed ease-in; // animate out behavior
  opacity: 0;
  transform: translate3d(-100%, 0, 0);
  visibility: hidden;
}

.elMBDrawer_Overlay {
  position: fixed;
  top:0;
  left:0;
  width: 100%;
  height: 100%;
  background: $brand-black;
  opacity: 0.5;
  z-index: 10;

  transition: all $drawer-animate-in-speed ease-out; // animate in behavior
}

.elMBDrawer_Overlay__Hidden {
  transition: all $drawer-animate-out-speed ease-in; // animate out behavior
  opacity: 0;
  visibility: hidden;
}

