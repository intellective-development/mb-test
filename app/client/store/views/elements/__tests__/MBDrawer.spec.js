
import React from 'react';

import MBDrawer from '../MBDrawer';

describe('MBDrawer', () => {
  it('renders', () => {
    expect(render(
      <MBDrawer open>
        <div>Fun!</div>
      </MBDrawer>
    )).toMatchSnapshot();
  });

  it('renders hidden state', () => {
    expect(render(
      <MBDrawer open={false}>
        <div>Fun!</div>
      </MBDrawer>
    )).toMatchSnapshot();
  });
});
