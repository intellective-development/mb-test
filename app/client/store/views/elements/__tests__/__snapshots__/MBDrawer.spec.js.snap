// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MBDrawer renders 1`] = `
<div>
  <div
    class="elMBDrawer_Overlay"
  />
  <div
    class="elMBDrawer"
  >
    <div>
      Fun!
    </div>
  </div>
</div>
`;

exports[`MBDrawer renders hidden state 1`] = `
<div>
  <div
    class="elMBDrawer_Overlay elMBDrawer_Overlay__Hidden"
  />
  <div
    class="elMBDrawer elMBDrawer__Hidden"
  >
    <div>
      Fun!
    </div>
  </div>
</div>
`;
